aptos-move/framework/aptos-framework/sources/account/account_abstraction.move

```diff
@@ -170,10 +181,17 @@ module aptos_framework::account_abstraction {
         func_info: FunctionInfo,
         signing_data: AbstractionAuthData,
     ): signer acquires DispatchableAuthenticator {
-        let func_infos = dispatchable_authenticator_internal(signer::address_of(&account));
+        let master_signer_addr = signer::address_of(&account);
+        let func_infos = dispatchable_authenticator_internal(master_signer_addr);
         assert!(ordered_map::contains(func_infos, &func_info), error::not_found(EFUNCTION_INFO_EXISTENCE));
         function_info::load_module_from_function(&func_info);
-        dispatchable_authenticate(account, signing_data, &func_info)
+        let returned_signer = dispatchable_authenticate(account, signing_data, &func_info);
+        // Returned signer MUST represent the same account address. Otherwise, it may break the invariant of Aptos blockchain!
+        assert!(
+            master_signer_addr == signer::address_of(&returned_signer),
```

Comment:
we should have a unit test or smoke test to check that transaction with different signer will be rejected (though not sure if move code even allows a way to return a different signer?

Reply:
users' code cannot call `create_storable_permissioned_handle` to create a permissioned signer. So now it is impossible to achieve this by 3rd party(non-0x1) AA module.

Reply:
Ok,  then it's fine without the test

Reply:
it's possible to generate a different signer by generating a `signer` in move with a `SignerCapability` or `ExendRef` to create an object/resource group signer

Reply:
I was thinking about p_signer only. Let me write a test for this case. Thanks for the info @khokho